syntax = "proto3";

//defines the FrequencyCalculator service and define the two functions
service FrequencyCalculator {

  rpc Calculate (TextToCount) returns (WordCount) {}

  rpc Combine (TextToCombine) returns (TotalCount) {}
}
//defines what the structure of the messages should be
message TextToCount {
  string text = 1;
}

//using the map structure, so that it resembles a dictionary structure
message WordCount {
  map<string,int64> WordOccurrence= 1;
}

message tuples {
  string key = 1;
  int64 value = 2;
}

//the repeated field, creates a list of the message tuples
message TextToCombine {
  repeated tuples allwords = 1;
}

message TotalCount {
  map<string,int64> WordOccurrence= 1;
}